tests/test_dse.py

```diff
@@ -475,6 +475,49 @@ def test_uncontracted_shape(self):
         op1(time_M=1)
         assert np.all(u.data == exp)
 
+    def test_uncontracted_shape_invariants(self):
+        """
+        Like `test_uncontracted_shape`, but now with some (outer-)Dimension-invariant
+        aliasing expressions.
+        """
+        grid = Grid(shape=(6, 6, 6))
+        x, y, z = grid.dimensions  # noqa
+        t = grid.stepping_dim
+
+        f = Function(name='f', grid=grid)
+        f.data_with_halo[:] =\
+            np.linspace(-1, 1, f.data_with_halo.size).reshape(*f.shape_with_halo)
+        u = TimeFunction(name='u', grid=grid, space_order=3)
+        u.data_with_halo[:] = 0.5
```

Comment:
any reason to add valeu to halo?

Reply:
not in this example, just copy-paste from the other tests. I can tweak it